pkg/gadgettracermanager/gadgettracermanager.go

```diff
@@ -344,8 +344,8 @@ func newServer(conf *Conf) (*GadgetTracerManager, error) {
 	containerEventFuncs := []pubsub.FuncNotify{}
 
 	if !conf.TestOnly {
-		if err := increaseRlimit(); err != nil {
-			return nil, fmt.Errorf("failed to increase memlock limit: %w", err)
+		if _, err := ebpf.RemoveMemlockRlimit(); err != nil {
+			return nil, err
```

Comment:
We can keep the error wrapping like previously in order to provide more context to the end-user in case of failure:
```suggestion
			return nil, fmt.Errorf("failed to remove memlock limit: %w", err)
```

Reply:
I removed it because cilium/ebpf already provides context:
https://github.com/cilium/ebpf/blob/4d5607ee1690d067b7f8befdd69b2bbee07e644b/internal/unix/types_linux.go#L213
```
		return nil, fmt.Errorf("failed to set memlock rlimit: %w", err)
```

Reply:
I just noticed that cilium/ebpf removed `ebpf.RemoveMemlockRlimit()` from its API: https://github.com/cilium/ebpf/pull/438

But it's ok because IG uses a previous version (see go.mod):
```
github.com/cilium/ebpf v0.6.3-0.20210910140648-4d5607ee1690
```

When IG updates its dependency to a newer version, we will need to use the new function `ebpf.RemoveMemlock()`. This comment is interesting:
https://github.com/cilium/ebpf/pull/438/files#diff-d4cbb04c5b92e360643b4373b4adedf77238268e07f192f9c888416bf7c484b6R79-R84

Reply:
> This comment is interesting:

So that's the reason why it was working fine on 5.11+.